interpreter/valid/match.ml

```diff
@@ -146,12 +153,16 @@ and match_global_type c a (GlobalType (t1, mut1)) (GlobalType (t2, mut2)) =
   | Immutable -> match_value_type c a t1 t2
   | Mutable -> eq_value_type c [] t1 t2
 
+and match_event_type c a (EventType (ft1, res1)) (EventType (ft2, res2)) =
+  eq_resumability c [] res1 res2 && match_func_type c [] ft1 ft2
```

Comment:
Shouldn't we pass `a` in place of `[]` here, even though it is ignored by both `eq_resumability` and `match_func_type`?

Reply:
Ah yes, we should for match_func_type. Though not for eq_res, since that is switching from subtyping to equivalence, so mustn't use the subtyping assumption. Tweaked.